src/bootstrap/src/core/build_steps/vendor.rs

```diff
@@ -46,12 +46,12 @@ pub(crate) struct Vendor {
     pub(crate) versioned_dirs: bool,
     /// The root directory of the source code.
     pub(crate) root_dir: PathBuf,
-    /// The target directory for storing vendored dependencies.
-    pub(crate) output_dir: PathBuf,
+    /// The target directory for storing vendored dependencies if different from root_dir.
+    pub(crate) output_dir: Option<PathBuf>,
 }
 
 impl Step for Vendor {
-    type Output = VendorOutput;
+    type Output = ();
```

Comment:
Why remove this? `GenerateCopyright` can still make use of the final path, instead of having to know about `VENDOR_DIR`, which is an implementation detail of the vendor step.

Reply:
`VendorOutput` only ever contained the data that should be written to `.cargo/config.toml`.

Reply:
I don't like overwriting the config.toml file on disk (as per my other comment), but if we did that, we can at least return the path to the vendor directory from the step, to avoid other steps reaching for `VENDOR_DIR`.